### PR TITLE
Improve logging, fix geth test

### PR DESCRIPTION
**Pull request by tudor-malene**
This PR is a result of debugging the very flaky geth test:
- made the logging consistent and easy to follow.
- slightly refactored the geth test according to the general pattern, and fix the duration 
- tweak the parameters of geth, as it turns out it takes a few seconds to publish a transaction

### CODE REVIEW

File: go/obscuronode/enclave/db/storage.go [hunk 1]
```diff
@@ -173,6 +173,9 @@ func (s *storageImpl) FetchBlockState(hash obscurocommon.L1RootHash) (*BlockStat
 }
 
 func (s *storageImpl) SetBlockState(hash obscurocommon.L1RootHash, state *BlockState) {
+	if state.Block.Hash() != hash {
+		panic("oops")
```
Comment (BedrockSquirrel) [inline go/obscuronode/enclave/db/storage.go:177]: Not sure if you meant to leave this in - worth making it more descriptive?

File: go/obscuronode/host/host.go [hunk 2]
```diff
@@ -368,21 +373,28 @@ func (a *Node) processBlocks(blocks []obscurocommon.EncodedBlock, interrupt *int
...
 		a.P2p.BroadcastRollup(nodecommon.EncodeRollup(result.ProducedRollup.ToRollup()))
 
-		obscurocommon.ScheduleInterrupt(a.cfg.GossipRoundDuration, interrupt, a.handleHeader(result))
+		obscurocommon.ScheduleInterrupt(a.cfg.GossipRoundDuration, interrupt, a.handleRoundWinner(result))
 	}
 }
 
-func (a *Node) handleHeader(result nodecommon.BlockSubmissionResponse) func() {
+func (a *Node) handleRoundWinner(result nodecommon.BlockSubmissionResponse) func() {
+	resultCopy := result
```
Comment (BedrockSquirrel) [inline go/obscuronode/host/host.go:381]: What does this copy achieve? (For my understanding, not because I think it's wrong)

Comment (tudor-malene) [reply go/obscuronode/host/host.go:381]: yeah. forgot to remove it. I wanted to make sure that the lambda is somehow working on the right result.

**Comment by tudor-malene** [comment]
Back to you @BedrockSquirrel

**Review by BedrockSquirrel — APPROVED** [review]
LGTM now

**Pull request merged**